File: Tutorials/ObjC/Example7/Example7/PlaylistDelegateProvider.m
//
//  PlaylistDelegateProvider.m
//  Example7
//
//  Created by Penthera on 7/11/19.
//  Copyright © 2019 dev. All rights reserved.
//

#import "PlaylistDelegateProvider.h"

@implementation PlaylistDelegateProvider

/*
 * This method is called by PlaylistManager when it needs next item in a Playlist
 */
-(VirtuosoPlaylistDownloadAssetItem*)assetForAssetID:(NSString *)assetID {
        
    VirtuosoAssetConfig* config = nil;
    if ([assetID isEqualToString:@"SEASON-1-EPISODE-2"])
    {
        config = [[VirtuosoAssetConfig alloc] initWithURL:@"https://virtuoso-demo-content.s3.amazonaws.com/bbb/season1/ep2/index.m3u8"
                                                  assetID:@"SEASON-1-EPISODE-2"
                                              description:@"Shadow Play"
                                                     type:kVDE_AssetTypeHLS];
    }
    else if ([assetID isEqualToString:@"SEASON-1-EPISODE-3"])
    {
        config = [[VirtuosoAssetConfig alloc] initWithURL:@"https://virtuoso-demo-content.s3.amazonaws.com/bbb/season1/ep3/index.m3u8"
                                                  assetID:@"SEASON-1-EPISODE-3"
                                              description:@"Aloha Adventure"
                                                     type:kVDE_AssetTypeHLS];
    }
    else
    {
        NSLog(@"Episode not found for assetID: %@", assetID);
    }
    if (config)
    {
        return [[VirtuosoPlaylistDownloadAssetItem alloc]initWithAsset:config];
    }
    else
    {
        return [[VirtuosoPlaylistDownloadAssetItem alloc]initWithOption:PlaylistDownloadOption_SkipToNext];
    }
}

@end

File: Tutorials/Swift/Example7/Example7/PlaylistDelegateProvider.swift
//
//  PlaylistDelegateProvider.swift
//  Example7
//
//  Created by Penthera on 7/1/19.
//  Copyright © 2019 Penthera. All rights reserved.
//

import Foundation


class PlaylistDelegateProvider : NSObject, VirtuosoPlaylistManagerDelegate {
    
    /*
     * This method is called by PlaylistManager when it needs next item in a Playlist
     */
    
    @objc func asset(forAssetID assetID: String) -> VirtuosoPlaylistDownloadAssetItem {
        
        var config: VirtuosoAssetConfig?
        
        switch assetID {
        case "SEASON-1-EPISODE-2":
            config = VirtuosoAssetConfig(url: "https://virtuoso-demo-content.s3.amazonaws.com/bbb/season1/ep2/index.m3u8",
                                         assetID: "SEASON-1-EPISODE-2",
                                         description: "Shadow Play",
                                         type: .vde_AssetTypeHLS)
            break
            
        case "SEASON-1-EPISODE-3":
            config = VirtuosoAssetConfig(url: "https://virtuoso-demo-content.s3.amazonaws.com/bbb/season1/ep3/index.m3u8",
                                         assetID: "SEASON-1-EPISODE-3",
                                         description: "Aloha Adventure",
                                         type: .vde_AssetTypeHLS)
            break

        case "SEASON-1-EPISODE-4":
            config = VirtuosoAssetConfig(url: "https://virtuoso-demo-content.s3.amazonaws.com/bbb/season1/ep4/index.m3u8",
                                         assetID: "SEASON-1-EPISODE-4",
                                         description: "Squirrelly",
                                         type: .vde_AssetTypeHLS)
            break

        case "SEASON-1-EPISODE-5":
            config = VirtuosoAssetConfig(url: "https://virtuoso-demo-content.s3.amazonaws.com/bbb/season1/ep5/index.m3u8",
                                         assetID: "SEASON-1-EPISODE-5",
                                         description: "Apples Raining",
                                         type: .vde_AssetTypeHLS)
            break

        case "SEASON-1-EPISODE-6":
            config = VirtuosoAssetConfig(url: "https://virtuoso-demo-content.s3.amazonaws.com/bbb/season1/ep6/index.m3u8",
                                         assetID: "SEASON-1-EPISODE-6",
                                         description: "Fighting Back",
                                         type: .vde_AssetTypeHLS)
            break

        default:
            print("Episode not found for assetID: : \(assetID)")
            break
        }
        
        guard let assetConfig = config else {
            return VirtuosoPlaylistDownloadAssetItem(option: .PlaylistDownloadOption_SkipToNext);
        }
        return VirtuosoPlaylistDownloadAssetItem(asset: assetConfig);
    }
    
}
